### PR TITLE
Allow graded and evaluated answers to be re-evaluated.

### LINKED ISSUE #1931: "Evaluate Answer" button does nothing for the user
Answers are only sent for autograding when they are in a 'submitted' state. However, users often want to force another evaluation of the answer, for instance when test cases have been updated (#1923).

Ignore the current status of the answer when calling `auto_grade!` on an answer.

This will also make it easier to fix #1923.

### PR DESCRIPTION
**Pull request by fonglh**
Gives instructors the flexibility to re-evaluate an answer with the
"Evaluate Answers" button on the submission edit page.

Will also allow answers to be automatically re-evaluated when
programming question test cases are modified after answer submission.

Fixes #1931, needed for #1923.

### CODE REVIEW

File: app/services/course/assessment/submission/auto_grading_service.rb [hunk 1]
```diff
@@ -24,9 +24,9 @@ def grade(submission)
 
   private
 
-  # Grades the answers in the provided submission which has not yet been graded and auto gradable.
+  # Grades the answers in the provided submission which are auto gradable.
   def grade_answers(submission)
-    auto_gradable_answers = ungraded_answers(submission).select do |answer|
+    auto_gradable_answers = submission.answers.select do |answer|
```
Comment (allenwq) [inline app/services/course/assessment/submission/auto_grading_service.rb]: `@submission.latest_answers` ?  otherwise all answers will be force re-graded

**Comment by allenwq** [comment]
@fonglh Need to update existing tests for this also.

**Comment by fonglh** [comment]
The existing tests didn't fail with this change though.

**Comment by allenwq** [comment]
Oh well, I think you also cannot test when the programming answers.

**Comment by weiqingtoh** [comment]
Is it possible to create a feature test to check if the evaluate code button is present for programming answers in the required state?

**Comment by fonglh** [comment]
The button checks for the state of the submission. It didn't use to do anything when answers were evaluated, so checking for the presence of the button is insufficient.

**Pull request merged**